File: RecycleTest/app/src/main/java/com/example/pch/recycletest/CustomAdapter.java
package com.example.pch.recycletest;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

public class CustomAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {
    private ArrayList<String> mList;
    public class CustomViewHolder extends RecyclerView.ViewHolder{
        private TextView text;
        public CustomViewHolder(View view){
            super(view);
            text = view.findViewById(R.id.something);
        }
    }

    public CustomAdapter(ArrayList<String> list){
        mList = list;
    }

    @NonNull
    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int viewType) {
        View view = LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.test_item2,viewGroup,false);
        CustomViewHolder viewHolder = new CustomViewHolder(view);
        return viewHolder;
    }

    @Override
    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder customViewHolder, int i) {
        if(customViewHolder instanceof CustomViewHolder) {
            ((CustomViewHolder)customViewHolder).text.setText(mList.get(i));
        }

    }

    @Override
    public int getItemCount() {
        return mList == null ? 0 : mList.size();
    }
}

File: background/app/src/main/java/com/example/pch/background/CustomAdapter.kt
package com.example.pch.background

import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import android.widget.Toast
import androidx.recyclerview.widget.RecyclerView
import kotlinx.android.synthetic.main.view_nationality.view.*

class CustomAdapter(private val nationalityList: MutableList<CustomView>) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {

    class CustomViewHolder(itemView: CustomView) : RecyclerView.ViewHolder(itemView) {
        fun onClick() {
            customView.nationality_korean.text ="${customView.nationality_korean.text}1"
            Log.d("custom","abc?")
        }

        val customView = itemView
        init {
            customView.setOnClickListener {
                onClick()
            }
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        //val view = LayoutInflater.from(parent.context).inflate(R.layout.view_nationality,parent,false)
        return CustomViewHolder(CustomView(parent.context).apply{
            layoutParams = ViewGroup.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT)
        })
    }

    override fun getItemCount(): Int {
        return nationalityList.size
    }

    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
        if(holder is CustomViewHolder) {
            holder.customView.run{
                nationality_korean.text = nationalityList[position].nationality_korean.text
                nationality_english.text = nationalityList[position].nationality_english.text
            }
        }
    }

}

interface ItemClickListener